test/sqlOutputContentProvider.test.ts

```diff
@@ -1,105 +1,160 @@
+import * as stubs from './stubs';
+import Constants = require('../src/models/constants');
+import vscode = require('vscode');
+import * as TypeMoq from 'typemoq';
+import assert = require('assert');
+
+suite('SqlOutputProvider Tests', () => {
+    let vscodeWrapper: TypeMoq.Mock<VscodeWrapper>;
+    let contentProvider: SqlOutputContentProvider;
+    let context: TypeMoq.Mock<vscode.ExtensionContext>;
+    let statusView: TypeMoq.Mock<StatusView>;
+
+    setup(() => {
+        vscodeWrapper = TypeMoq.Mock.ofType(VscodeWrapper);
+        context = TypeMoq.Mock.ofType(stubs.TestExtensionContext);
+        context.object.extensionPath = '';
+        statusView = TypeMoq.Mock.ofType(StatusView);
+        contentProvider = new SqlOutputContentProvider(context.object, statusView.object);
+        contentProvider.setVscodeWrapper(vscodeWrapper.object);
+
+    });
+
+    test('Correctly outputs the new result pane view column', done => {
+        function setSplitPaneSelectionConfig(value: string): void {
+            let configResult: {[key: string]: any} = {};
+            configResult[Constants.configSplitPaneSelection] = value;
+            let config = stubs.createWorkspaceConfiguration(configResult);
+            vscodeWrapper.setup(x => x.getConfiguration(TypeMoq.It.isAny()))
+            .returns(x => {
+                return config;
+            });
+        }
+
+        function setCurrentEditorColumn(column: number): void {
+            vscodeWrapper.setup(x => x.activeTextEditor)
+            .returns(x => {
+                let editor: vscode.TextEditor = new stubs.TestTextEditor();
+                editor.viewColumn = column;
+                return editor;
+            });
+        }
+
+        class Case {
+            position: number;
+            config: string;
+            expectedColumn: number;
+        }
+
+        let cases: Case[] = [
+            {position: 1, config: 'next', expectedColumn: 2},
+            {position: 2, config: 'next', expectedColumn: 3},
+            {position: 3, config: 'next', expectedColumn: 3},
+            {position: 1, config: 'same', expectedColumn: 1},
+            {position: 2, config: 'same', expectedColumn: 2},
+            {position: 3, config: 'same', expectedColumn: 3},
+            {position: 1, config: 'last', expectedColumn: 3},
+            {position: 2, config: 'last', expectedColumn: 3},
+            {position: 3, config: 'last', expectedColumn: 3}
+        ];
+
+        try {
+            cases.forEach((c: Case) => {
+                setSplitPaneSelectionConfig(c.config);
+                setCurrentEditorColumn(c.position);
+
+                let resultColumn = contentProvider.newResultPaneViewColumn();
+                assert.equal(resultColumn, c.expectedColumn);
+            });
+
+            done();
+        } catch (err) {
+            done(new Error(err));
+        }
+    });
+
+});
+
+
+// TODO: rewrite all the outputprodiver handle tests (old ones kept for reference)
```

Comment:
is there an issue tracking it? if so what's the issue #? also typo `outputprovider`